docs/src/pages/system/advanced/advanced.md

```diff
@@ -0,0 +1,21 @@
+# Advanced
+
+<p class="description">Here you can find examples of how you can use the system on your custom components.</p>
+
+## Adding the `sx` prop on your custom components
+
+The easiest way to support the `sx` prop on your custom components is to use the `experimentalStyled()` utility that comes from `@material-ui/core/styles` for creating your custom components. Here is an example of how it can be used:
+
+{{"demo": "pages/system/advanced/ExperimentalStyledDemo.js", "defaultCodeOpen": true}}
+
+## Use the `unstable_styleFuntionSx` utility
+
+The `experimentalStyled()` offers other functionalities, like access to the default theme, as well as access to the theme overrides and variants. If you would not like to use these utilities, you may use the lighter version of the utility for adding the `sx` utility on your custom components - the `unstable_styleFunctionSx` utility.
+
+{{"demo": "pages/system/advanced/StyleFunctionSxDemo.js"}}
```

Comment:
We might want to focus on `@material-ui/system`. I would challenge any import from `@material-ui/core`.

```suggestion
## Adding the `sx` prop on your custom components

The `unstable_styleFunctionSx` utility adds the support for the `sx` in your own components.

{{"demo": "pages/system/advanced/StyleFunctionSxDemo.js"}}

### Built-in `sx` from the core

If you are already using the Material-UI's `styled()` wrapper, the easiest way to support the `sx` prop on your custom components is to use the `experimentalStyled()` utility that comes from `@material-ui/core/styles` for creating your custom components. Here is an example of how it can be used:

{{"demo": "pages/system/advanced/ExperimentalStyledDemo.js", "defaultCodeOpen": true}}
```

Reply:
It felt weird to me too that I am using the `experimentalStyled()` on this guide, but not sure what is the best place for it. Regarding the order, I agree. 

Not sure if the core is the best place for exporting the `experimentalStyled()` but it definitely depends on lots of core things, especially the theme structure that we use..